Fix aarch64/armv7/ppc64 feature detection for systems with AT_HWCAP != 16
c0697e73a19b46fb389aeb5fe14a92b2275b5301

Index: src/lib/utils/os_utils.cpp
--- src/lib/utils/os_utils.cpp.orig
+++ src/lib/utils/os_utils.cpp
@@ -106,6 +106,26 @@ uint32_t OS::get_process_id()
 #endif
    }
 
+unsigned long OS::auxval_hwcap() {
+#if defined(AT_HWCAP)
+   return AT_HWCAP;
+#else
+   // If the value is not defined in a header we can see,
+   // but auxval is supported, return the Linux/Android value
+   return (OS::has_auxval()) ? 16 : 0;
+#endif
+}
+
+unsigned long OS::auxval_hwcap2() {
+#if defined(AT_HWCAP2)
+   return AT_HWCAP2;
+#else
+   // If the value is not defined in a header we can see,
+   // but auxval is supported, return the Linux/Android value
+   return (OS::has_auxval()) ? 26 : 0;
+#endif
+}
+
 unsigned long OS::get_auxval(unsigned long id)
    {
 #if defined(BOTAN_TARGET_OS_HAS_GETAUXVAL)
